security/cmd/node_agent_k8s/main.go

```diff
@@ -86,30 +97,56 @@ var (
 	}
 )
 
-func init() {
-	caProvider := os.Getenv(caProvider)
-	if caProvider == "" {
-		log.Error("CA Provider is missing")
-		os.Exit(1)
+func newSecretCache(serverOptions sds.Options) (workloadSecretCache, gatewaySecretCache *cache.SecretCache) {
+	if serverOptions.EnableWorkloadSDS {
+		wSecretFetcher, err := secretfetcher.NewSecretFetcher(false, serverOptions.CAEndpoint, serverOptions.CAProviderName, true)
+		if err != nil {
+			log.Errorf("failed to create secretFetcher for workload proxy: %v", err)
+			os.Exit(1)
+		}
+		cacheOptions.TrustDomain = serverOptions.TrustDomain
+		cacheOptions.Plugins = sds.NewPlugins(serverOptions.PluginNames)
+		workloadSecretCache = cache.NewSecretCache(wSecretFetcher, sds.NotifyProxy, cacheOptions)
+	} else {
+		workloadSecretCache = nil
 	}
 
-	caAddr := os.Getenv(caAddress)
-	if caAddr == "" {
-		log.Error("CA Endpoint is missing")
-		os.Exit(1)
+	if serverOptions.EnableIngressGatewaySDS {
+		gSecretFetcher, err := secretfetcher.NewSecretFetcher(true, serverOptions.CAEndpoint, serverOptions.CAProviderName, true)
+		if err != nil {
+			log.Errorf("failed to create secretFetcher for gateway proxy: %v", err)
+			os.Exit(1)
+		}
+		cacheOptions.TrustDomain = serverOptions.TrustDomain
```

Comment:
trustdomain and plugins is not used by ingress case today, so probably could be removed.